src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonperson/PrisonPersonService.kt

```diff
@@ -57,4 +63,47 @@ class PrisonPersonService(
         ?.movementTime
         ?: bookingEndDate
     }
+
+  fun upsertPhysicalAttributes(offenderNo: String, request: UpsertPhysicalAttributesRequest) {
+    val booking = bookingRepository.findLatestByOffenderNomsId(offenderNo)
+      ?: throw NotFoundException("No latest booking found for $offenderNo")
+
+    val physicalAttributes = booking.physicalAttributes.find { it.id.sequence == 1L }
+      ?: OffenderPhysicalAttributes(id = OffenderPhysicalAttributeId(booking, 1L))
+
+    physicalAttributes.setWeightInKilograms(request.weight)
+    physicalAttributes.setHeightInCentimetres(request.height)
+    offenderPhysicalAttributesRepository.save(physicalAttributes)
```

Comment:
I guess you don't need the `save` given you are amending the booking that was just retrieved ?

Reply:
It could be a new record - see line 72.